bot/cogs/grades_cog/grades_cog.py

```diff
@@ -170,108 +233,110 @@ async def prof(self, ctx, *, prof: str):
 
             val = f'```A: {A}\nB: {B}\nC: {C}\nD: {D}\nF: {F}\nW: {W}```'
             embed.add_field(name=f"{row.name}'s distribution", value=val)
-            embed.add_field(name='Total Number of Classes Analyzed', value=len(df), inline=False)
-            embed.add_field(name='Explanation',value=f'Run `{await self.bot.current_prefix(ctx)}help grades` for more information on this command')
+            embed.add_field(name='Total Number of Classes Analyzed', value=str(len(df)), inline=False)
```

Comment:
what does thsi str cast do?

Reply:
PyCharm was complaining that it was expecting a string there, not an int

Reply:
... huh? df is a dataframe, calling len on a dataframe gives you the number of rows

Reply:
Yes. When you run `len(df)`, the output is an integer: the number of rows in the dataframe. Discord's `embed.add_field` expects that the `value` parameter be a string, not an integer. So this type conversion corrects the value to the expected format. Described in the [API](https://discordpy.readthedocs.io/en/stable/api.html#discord.Embed.add_field):

> add_field(*, name, value, inline=True)
> Adds a field to the embed object.
>
> This function returns the class instance to allow for fluent-style chaining.
>
> Parameters
> name (str) – The name of the field.
> **value (str) – The value of the field.**
> inline (bool) – Whether the field should be displayed inline.

Reply:
wow... im really fucking tired right now lmao you are correct and i am stupid